apps/sms/js/activity_handler.js

```diff
@@ -558,7 +558,15 @@ var ActivityHandler = {
     //
     var params = Utils.params(message.imageURL);
 
+    // When notification is removed from notification tray, notification system
+    // message will still be fired, but "clicked" property will be equal to
+    // false. When user clicks on notification we'll get two system messages,
+    // first to notify app that notification is clicked and the second one to
+    // indicate that notification is removed from notification tray.
```

Comment:
> maybe go to a new line before "When", and explain we get 2 system messages because we remove the notification from SMS?

Okay!

> Also add a reference to the bug # where we'll remove this close event ?

Mmm, not sure that we have one, afaik Michael was talking about some non-existent-yet bug where notification guys are going to make Gecko Notification API conformant with the latest spec (including removing of "show" and "close" events). But I can file one for "onclose" only if you prefer.